test/games/strategy/triplea/delegate/BattleOrderTest.java

```diff
@@ -0,0 +1,112 @@
+    final Territory iceland = m_data.getMap().getTerritory("Iceland");
+    final Territory sz122 = m_data.getMap().getTerritory("122 Sea Zone");
+    final PlayerID british = GameDataTestUtil.british(m_data);
+    final PlayerID germans = GameDataTestUtil.germans(m_data);
+    final ITestDelegateBridge bridge = getDelegateBridge(germans);
+    bridge.setStepName("germansBattle");
+    delegate.setDelegateBridgeAndPlayer(bridge);
+    delegate.start();
+    System.out.println(iceland.getOwner().getName());
+/*
+    // Find facilities in territory
+    for (final Unit target : wgermany.getUnits().getUnits()) {
+      switch (target.getType().getName()) {
+        case "airfield":
+          airfield = (TripleAUnit) target;
+          break;
+        case "harbour":
+          harbour = (TripleAUnit) target;
+          break;
+        case "factory_major":
+          factory = (TripleAUnit) target;
+          break;
+      }
+    }
+
+    targets.put(airfield, new HashSet<>(Collections.singleton(tacBomber1)));
+    targets.put(harbour, new HashSet<>(Collections.singleton(tacBomber2)));
+    targets.put(factory, new HashSet<>(Collections.singleton(stratBomber)));
+    final ITestDelegateBridge bridge = getDelegateBridge(british);
+    tracker.addBattle(new RouteScripted(wgermany), attackers, true, british, bridge, null, null, targets, true);
+
+    battle.addAttackChange(m_data.getMap().getRoute(uk, wgermany),
+        uk.getUnits().getMatches(Matches.UnitIsStrategicBomber), null);
+    // addTo(wgermany, uk.getUnits().getMatches(Matches.UnitIsStrategicBomber));
+    tracker.getBattleRecords().addBattle(british, battle.getBattleID(), wgermany, battle.getBattleType());
+    // aa guns rolls 1,3,2 first one hits, remaining bombers roll 1 dice each
+    bridge.setRandomSource(new ScriptedRandomSource(new int[] {1, 3, 2, 5, 4}));
+    final InvocationHandler handler = new InvocationHandler() {
+      @Override
+      public Object invoke(final Object proxy, final Method method, final Object[] args) throws Throwable {
+        return null;
+      }
+    };
+    final ITripleAPlayer player = (ITripleAPlayer) Proxy
+        .newProxyInstance(Thread.currentThread().getContextClassLoader(),
+            TestUtil.getClassArrayFrom(ITripleAPlayer.class), handler);
+    bridge.setRemote(player);
+    battle.fight(bridge);
+    final int airfieldDmg = airfield.getUnitDamage();
+    final int harbourDmg = harbour.getUnitDamage();
+    final int factoryDmg = factory.getUnitDamage();
+    System.out.format("airf: %d harb: %d fact: %d\n", airfieldDmg, harbourDmg, factoryDmg);
+    // targets dice is 4, so damage is 1 + 4 = 5
+    // bomber 2 hits at 2, so damage is 3, for a total of 8
+    // Changed to match StrategicBombingRaidBattle changes
+    // All tests fail. Remove to get past automated testing. Change works in normal game. Not sure why it doesn't work
+    // here. Probably some problem with the test.
+*/
+    assertTrue( iceland.getOwner() == germans );
```

Comment:
What it tests is that the amphibious assault on Iceland actually happened automatically.